tests/build-test-report.js

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/env node
+const os = require('os');
+const path = require('path');
+const yargs = require('yargs');
+const fs = require('fs');
+const execSync = require('child_process').execSync;
+const spawn = require('child_process').spawn;
+const chalk = require('chalk');
+const runner = require('./runner');
+
+
+const TESTS_DIRECTORY = path.resolve(__dirname, 'test262');
+const DEFAULT_TEST_RESULTS_FILE = 'test-results-new.json';
+const DEFAULT_VERBOSE_TEST_RESULTS_FILE = 'test-results-new.verbose.json';
+const SAVED_RESULTS_FILE = path.resolve(__dirname, 'test-results.json');
+
+const argv = yargs
+  .usage(`Usage: $0 [options] [${DEFAULT_TEST_RESULTS_FILE}]`)
+
+  .alias('d', 'diff')
+  .describe('d', 'diff against existing test results. Returns exit code 1 if there are changes.')
+
+  .alias('r', 'run')
+  .describe('r', 'generate new test results')
+
+  .alias('s', 'save')
+  .describe('s', 'save the results')
+
+  .alias('t', 'threads')
+  .describe('t', '# of threads to use')
+  .nargs('t', 1)
+  .default('t', os.cpus().length)
+
+  .alias('v', 'verbose')
+
+  .alias('g', 'glob')
+  .describe('g', 'glob of tests to run')
+  .nargs('g', 1)
+
+  .alias('o', 'out')
+  .describe('o', 'Directory to dump compiled test files to')
+  .nargs('o', 1)
+
+  .help('h')
+  .alias('h', 'help')
+  .argv;
+
+const RESULTS_FILE = argv._[0] || path.resolve(__dirname, DEFAULT_TEST_RESULTS_FILE);
+const VERBOSE_RESULTS_FILE = path.resolve(__dirname, DEFAULT_VERBOSE_TEST_RESULTS_FILE);
+
+function downloadTestsIfNecessary() {
+  if (!fs.existsSync(TESTS_DIRECTORY)) {
+    console.log("Downloading test262 test suite");
+    execSync(`git clone https://github.com/tc39/test262.git --depth 1 ${TESTS_DIRECTORY}`);
+  }
+}
+
+function saveResults(results) {
+  console.log('Saving results for future comparison...');
+  results = results.map(test => ({
+    file: test.file,
+    attrs: test.attrs,
+    result: test.result,
+  }))
+  results.sort((a,b) => a.file < b.file ? -1 : a.file === b.file ? 0 : 1);
+  fs.writeFileSync(SAVED_RESULTS_FILE, JSON.stringify(results, null, 2));
+}
+
+function runTests(outputFilePath, verboseOutputFilePath) {
+  downloadTestsIfNecessary();
+  console.log(`running tests with ${argv.threads/1} threads...`);
```

Comment:
Just a shortcut for casting nulls and empty strings to 0.

```js
null/1;  // => 0
''/1; // => 0
```

But I think it's actually not necessary since a default is already set... will get rid of it.